src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -653,11 +654,10 @@ namespace Microsoft.FSharp.Core
 
             let inline Array2DZeroCreate (n:int) (m:int) = (# "newarr.multi 2 !0" type ('T) n m : 'T[,] #)
             let GetArray2DSub (src: 'T[,]) src1 src2 len1 len2 =
-                let len1 = (if len1 < 0 then 0 else len1)
-                let len2 = (if len2 < 0 then 0 else len2)
-                let dst = Array2DZeroCreate len1 len2
-                for i = 0 to len1 - 1 do
-                    for j = 0 to len2 - 1 do
+                let lenN, lenM = if len1 <= 0 || len2 <= 0 then 0, 0 else len1, len2
```

Comment:
Consider `struct(lenN, lenM)` to avoid boxing since these are primitive types

Reply:
Update: Seems like we can't use struct tuples in prim-types here. We investigated trying to optimize it by using outrefs, but poking around with the JIT assembly created by inlined functions reveals that the compiler/CLR is probably smart enough to not make the tuple, and hence returning a reference tuple is probably fine here.